Master adp5061 support misc

stefpopa opened the pull request:
This set of patches adds four new features:
1. enable/disable battery charging
2. enable/disable charging voltage limit
3. read the charger status 
4. read the battery status

The first two features are not configurable via the power_supply properties, therefore access via sysfs was provided to examine and modify these attributes on the fly.

stefpopa commented:
changelog v1 -> v2:
* added another patch for setting and reading the values of the termination current

commodo commented on drivers/power/supply/adp5061.c:
```diff
@@ -570,11 +574,69 @@ static const struct power_supply_desc adp5061_desc = {
+
+	ret = regmap_update_bits(st->regmap, ADP5061_FUNC_SET_1,
+				 ADP5061_FUNC_SET_1_EN_CHG_MSK,
+				 ADP5061_FUNC_SET_1_EN_CHG_MODE(chg_en));
```
curios: would it be an idea to just sanitize the value here instead of limiting to 0 / 1 ?
maybe something like this `ADP5061_FUNC_SET_1_EN_CHG_MODE(!!chg_en));`
that way, non-zero values enable and zero disables ; so, this can be somewhat more flexible

stefpopa replied:
I agree

commodo commented on drivers/power/supply/adp5061.c:
```diff
@@ -437,6 +437,41 @@ static int adp5061_get_chg_type(struct adp5061_state *st,
+		break;
+	default:
+		val->intval = POWER_SUPPLY_STATUS_UNKNOWN;
+
```
nitpick: un-needed empty line

commodo commented on drivers/power/supply/adp5061.c:
```diff
@@ -619,11 +623,60 @@ static int adp5061_set_charging_enabled(struct device *dev,
+
+	ret = regmap_update_bits(st->regmap, ADP5061_FUNC_SET_2,
+			ADP5061_FUNC_SET_2_EN_CHG_VLIM_MSK,
+			ADP5061_FUNC_SET_2_EN_CHG_VLIM_MODE(chg_vlim_en));
```
same comment here about allowing non-zero + zero values here

stefpopa commented:
changelog v2->v3:
* removed one extra blank line
* modified adp5061_set_chg_vlim_enabled() and adp5061_set_charging_enabled() such that non-zero values enable and zero disables.

lclausen-adi commented on drivers/power/supply/adp5061.c:
```diff
@@ -570,11 +701,107 @@ static const struct power_supply_desc adp5061_desc = {
+	return count;
+}
+
+static DEVICE_ATTR(charging_enabled, 0644, adp5061_get_charging_enabled,
```
Any custom property needs to be documented in Documentation/ABI/testing/sysfs-class-power-adp5061.

stefpopa commented:
changelog v3->v4:
* documented charging_enabled and charging_vlim_enabled custom properties in Documentation/ABI/testing/sysfs-class-power-adp5061.

commodo commented:
if no objections, will merge this tomorrow, later in the afternoon
